builder/main.py

```
@@ -1017,43 +1159,163 @@ def download_littlefs(target, source, env):
+        
+        # Check for FAT boot sector signature (0x55AA at offset 510-511)
+        # Note: For 4096-byte sectors, this might be at different offsets
+        boot_signature_512 = fs_data[510:512]
+        boot_signature_4096 = fs_data[4094:4096] if len(fs_data) >= 4096 else b''
+        
+        print(f"Debug: Boot signature @ 510: {boot_signature_512.hex()}")
+        print(f"Debug: Boot signature @ 4094: {boot_signature_4096.hex()}")
+        
+        # FFat uses standard 512-byte boot sector even with 4096-byte logical sectors
+        if boot_signature_512 != b'\x55\xaa':
+            print(f"Warning: Boot sector signature not found at offset 510")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)